pkg/record/interface.go

```diff
@@ -48,12 +48,28 @@ func (m JSONMarshaller) GetExtension() string {
 	return "json"
 }
 
+type gatherStatusReport struct {
+	Name    string        `json:"name"`
+	Elapsed time.Duration `json:"elapsed"`
+	Report  int           `json:"report"`
+	Errors  []error       `json:"errors"`
+}
+
 // Collect is a helper for gathering a large set of records from generic functions.
 func Collect(ctx context.Context, recorder Interface, bulkFns ...func() ([]Record, []error)) error {
 	var errors []string
+	var gatherReport []interface{}
 	for _, bulkFn := range bulkFns {
-		klog.V(5).Infof("Gathering %s", runtime.FuncForPC(reflect.ValueOf(bulkFn).Pointer()).Name())
+		gatherName := runtime.FuncForPC(reflect.ValueOf(bulkFn).Pointer()).Name()
+		klog.V(5).Infof("Gathering %s", gatherName)
+
+		start := time.Now()
 		records, errs := bulkFn()
+		elapsed := time.Now().Sub(start).Truncate(time.Millisecond)
+
+		klog.V(4).Infof("Gather %s took %s to process %d records", gatherName, elapsed, len(records))
```

Comment:
Why log level 4 here and 5 above? I would do 5 for both. It also seems to me little bit redundant to log the time when we have it in the archive.

Reply:
If the recorder fails, you have the logs. Maybe change to 5 is fine, but I was keeping it as the one that measure the execution time of the periodic:

```
klog.V(4).Infof("Periodic gather %s completed in %s", name, time.Now().Sub(start).Truncate(time.Millisecond))
```

Change to 5 or not?

Reply:
OK good point. Thanks. Well it's true it's minor but I would be for 5.